flink-runtime/src/main/java/org/apache/flink/runtime/operators/sort/UnilateralSortMerger.java

```diff
@@ -212,9 +215,39 @@ protected UnilateralSortMerger(MemoryManager memoryManager, List<MemorySegment>
 			TypeSerializerFactory<E> serializerFactory, TypeComparator<E> comparator,
 			int numSortBuffers, int maxNumFileHandles,
 			float startSpillingFraction, boolean noSpillingMemory, boolean handleLargeRecords,
-			boolean objectReuseEnabled)
-	throws IOException
-	{
+			boolean objectReuseEnabled) throws IOException {
+		this (
+			memoryManager,
+			memory,
+			ioManager,
+			input,
+			parentTask,
+			serializerFactory,
+			comparator,
+			numSortBuffers,
+			maxNumFileHandles,
+			startSpillingFraction,
+			noSpillingMemory,
+			handleLargeRecords,
+			objectReuseEnabled,
+			new DefaultInMemorySorterFactory<>(serializerFactory, comparator, THRESHOLD_FOR_IN_PLACE_SORTING));
+	}
+
+	protected UnilateralSortMerger(
```

Comment:
`@VisibleForTesting`?

Reply:
why? we might config which `InMemorySorterFactory` to use if needed. semantically it's no need to be `@VisibleForTesting`

Reply:
I agree with @TisonKun